configure.ml

```diff
@@ -45,16 +65,34 @@ let () =
       , Arg.String (set_dir etcdir)
       , "DIR where configuration files are installed for the default build \
          context" )
+    ; ( "--libexecdir"
+      , Arg.String (set_dir libexecdir)
+      , "DIR where files for the libexec_root section are installed for the \
+         default build context. Use --libdir if specified" )
+    ; ( "--datadir"
+      , Arg.String (set_dir datadir)
+      , "DIR where files for the share_root section are installed for the \
+         default build context" )
     ]
   in
   let anon s = bad "Don't know what to do with %s" s in
   Arg.parse (Arg.align args) anon
     "Usage: ocaml configure.ml [OPTIONS]\nOptions are:";
-  let oc = open_out "src/dune_rules/setup.ml" in
+  (match !libexecdir with
+  | None -> libexecdir := !library_destdir
+  | Some _ -> ());
+  let oc = open_out out in
   let pr fmt = fprintf oc (fmt ^^ "\n") in
-  pr "let library_path    = %s" (option (list string) !library_path);
-  pr "let library_destdir = %s" (option string !library_destdir);
-  pr "let mandir = %s" (option string !mandir);
-  pr "let docdir = %s" (option string !docdir);
-  pr "let etcdir = %s" (option string !etcdir);
+  pr "open! Dune_engine\n";
+  pr "let library_path = %s\n" ((list string) !library_path);
+  pr "let roots =\n  Install.Section.Paths.Roots.";
+  pr "    { lib_root = %s" (option string !library_destdir);
```

Comment:
`{ Install.Section.Paths.Roots.lib_root ...` to be consistent with the rest of our code.